packages/itwinui-react/src/core/ButtonGroup/ButtonGroup.tsx

```diff
@@ -203,40 +210,50 @@ const OverflowGroup = React.forwardRef((props, forwardedRef) => {
         },
         props.className,
       )}
-      ref={useMergedRefs(forwardedRef, overflowRef)}
+      ref={forwardedRef}
     >
-      {(() => {
-        if (!(visibleCount < items.length)) {
-          return items;
-        }
-
-        const overflowStart =
-          overflowPlacement === 'start'
-            ? items.length - visibleCount
-            : visibleCount - 1;
-
-        return (
-          <>
-            {overflowButton &&
-              overflowPlacement === 'start' &&
-              overflowButton(overflowStart)}
-
-            {overflowPlacement === 'start'
-              ? items.slice(overflowStart + 1)
-              : items.slice(0, Math.max(0, overflowStart))}
-
-            {overflowButton &&
-              overflowPlacement === 'end' &&
-              overflowButton(overflowStart)}
-          </>
-        );
-      })()}
-    </BaseGroup>
+      <OverflowGroupContent
+        overflowButton={overflowButton}
+        overflowPlacement={overflowPlacement}
+        items={items}
+      />
+    </OverflowContainer>
   );
-}) as PolymorphicForwardRefComponent<
-  'div',
-  Pick<
-    ButtonGroupProps,
-    'children' | 'orientation' | 'overflowButton' | 'overflowPlacement'
-  >
->;
+}) as PolymorphicForwardRefComponent<'div', OverflowGroupProps>;
+
+// ----------------------------------------------------------------------------
+
+type OverflowGroupContentProps = Pick<
+  OverflowGroupProps,
+  'overflowButton' | 'overflowPlacement'
+> & {
+  items: ReturnType<typeof React.Children.toArray>;
+};
+
+const OverflowGroupContent = (props: OverflowGroupContentProps) => {
+  const { overflowButton, overflowPlacement, items } = props;
+  const { visibleCount } = useOverflowContainerContext();
+
+  const overflowStart =
+    overflowPlacement === 'start'
+      ? items.length - visibleCount
+      : visibleCount - 1;
+
+  return !(visibleCount < items.length) ? (
+    items
```

Comment:
nit: Can we use `if` conditions (with early `return`s) instead of ternaries for better readability?